client/my-sites/domains/domain-management/settings/cards/site-redirect-card.jsx

```diff
@@ -0,0 +1,228 @@
+import FormButton from 'calypso/components/forms/form-button';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+import { withoutHttp } from 'calypso/lib/url';
+import { SITE_REDIRECT } from 'calypso/lib/url/support';
+import { domainManagementSiteRedirect } from 'calypso/my-sites/domains/paths';
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+} from 'calypso/state/analytics/actions';
+import {
+	closeSiteRedirectNotice,
+	fetchSiteRedirect,
+	updateSiteRedirect,
+} from 'calypso/state/domains/site-redirect/actions';
+import { getSiteRedirectLocation } from 'calypso/state/domains/site-redirect/selectors';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { fetchSiteDomains } from 'calypso/state/sites/domains/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+const noticeOptions = {
+	duration: 5000,
+	id: `site-redirect-update-notification`,
+};
+
+class SiteRedirectCard extends Component {
+	static propTypes = {
+		location: PropTypes.object.isRequired,
+		redesign: PropTypes.bool,
+		selectedDomainName: PropTypes.string.isRequired,
+		selectedSite: PropTypes.object.isRequired,
+	};
+
+	state = {
+		redirectUrl: this.props.location.value,
+	};
+
+	componentDidMount() {
+		this.props.fetchSiteRedirect( this.props.selectedSite.domain );
+	}
+
+	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
+	UNSAFE_componentWillReceiveProps( nextProps ) {
+		if ( this.props.location.value !== nextProps.location.value ) {
+			this.setState( {
+				redirectUrl: nextProps.location.value,
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		this.closeRedirectNotice();
+	}
+
+	closeRedirectNotice = () => {
+		this.props.closeSiteRedirectNotice( this.props.selectedSite.domain );
+	};
+
+	handleChange = ( event ) => {
+		const redirectUrl = withoutHttp( event.target.value );
+
+		this.setState( { redirectUrl } );
+	};
+
+	handleClick = () => {
+		this.props
+			.updateSiteRedirect( this.props.selectedSite.domain, this.state.redirectUrl )
+			.then( ( success ) => {
+				this.props.recordUpdateSiteRedirectClick(
+					this.props.selectedDomainName,
+					this.state.redirectUrl,
+					success
+				);
+
+				if ( success ) {
+					this.props.fetchSiteDomains( this.props.selectedSite.ID );
+					this.props.fetchSiteRedirect( this.state.redirectUrl.replace( /\/+$/, '' ).trim() );
+
+					page(
+						domainManagementSiteRedirect(
+							this.props.selectedSite.slug,
+							this.state.redirectUrl.replace( /\/+$/, '' ).trim(),
+							this.props.currentRoute
+						)
+					);
+
+					this.props.successNotice(
+						this.props.translate( 'Site redirect updated successfully.' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Your site redirect was updated successfully.' )` *ES Score: 7*